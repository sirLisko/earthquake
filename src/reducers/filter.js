const initState = {};

export const FILTER_SET = 'FILTER_SET';
export const FILTER_CLEAR = 'FILTER_CLEAR';

export const setFilter = data => ({
  type: FILTER_SET,
  payload: data,
});

export const clearFilter = data => ({
  type: FILTER_CLEAR,
});

export const getCurrentFilter = state => state.filter;
export const getCurrentFilterName = state => state.filter.name;

export default (state = initState, action) => {
  switch (action.type) {
    case FILTER_CLEAR:
      return {};
    case FILTER_SET:
      return { ...state, ...action.payload };
    default:
      return state;
  }
};
