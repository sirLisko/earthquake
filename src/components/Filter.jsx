import React, { useState, useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import PropTypes from 'prop-types';

import styled from '@emotion/styled/macro';

import { setFilter, getCurrentFilterName } from 'reducers/filter';
import { StyledButton } from 'styles/basic';
import theme from 'styles/theme';

const StyledInput = styled.input`
  font-size: 1rem;
  -webkit-border-radius: 0;
  -webkit-appearance: none;
  border: 0;
  border-bottom: 2px solid ${theme.colors.accent};
  padding: 0.5rem 1rem;
  margin: 0 0 1re;
  background: #f2f2f2;
`;

const Filter = ({ name }) => {
  const dispatch = useDispatch();
  const filterSelected = useSelector(getCurrentFilterName);
  const [value, setValue] = useState('');
  useEffect(() => {
    filterSelected !== name && setValue('');
  }, [filterSelected, name]);
  return (
    <form
      onSubmit={e => {
        e.preventDefault();
        dispatch(setFilter({ name, value }));
      }}
    >
      <StyledInput
        type="text"
        value={value}
        placeholder={`set ${name}`}
        onChange={({ target }) => setValue(target.value)}
      />
      <StyledButton type="submit">set</StyledButton>
    </form>
  );
};

export default Filter;

Filter.propTypes = {
  name: PropTypes.string.isRequired,
};
