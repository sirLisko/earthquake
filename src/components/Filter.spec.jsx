import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { mount } from 'enzyme';

import * as filter from 'reducers/filter';

import Filter from './Filter';

const mockStore = configureStore([]);

describe('Filter Component', () => {
  let wrapper;
  const emptyStore = mockStore({ filter: {} });

  beforeEach(() => {
    jest.spyOn(filter, 'setFilter');
    wrapper = mount(
      <Provider store={emptyStore}>
        <Filter name="foo" />
      </Provider>,
    );
  });

  it('should render properly', () => {
    expect(wrapper.find('Filter')).toMatchSnapshot();
  });

  it('should dispatch the correct action', () => {
    const fakeEvent = {
      preventDefault: jest.fn(),
      target: {
        value: 'foo',
      },
    };
    wrapper.find('form').simulate('submit', fakeEvent);
    expect(filter.setFilter).toHaveBeenCalled();
    expect(filter.setFilter).toHaveBeenCalledWith({ name: 'foo', value: '' });
  });
});
