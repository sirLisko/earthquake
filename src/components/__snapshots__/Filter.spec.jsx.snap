// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter Component should render properly 1`] = `
<Filter
  name="foo"
>
  <form
    onSubmit={[Function]}
  >
    <StyledInput
      onChange={[Function]}
      placeholder="set foo"
      type="text"
      value=""
    >
      <input
        className="css-k5oqm0-StyledInput ei529xs0"
        onChange={[Function]}
        placeholder="set foo"
        type="text"
        value=""
      />
    </StyledInput>
    <StyledButton
      type="submit"
    >
      <button
        className="css-1a38x1d-StyledButton enac65f0"
        type="submit"
      >
        set
      </button>
    </StyledButton>
  </form>
</Filter>
`;
